File: apps/ai-app/app/router/root.py
from app.core.database import DepDB, check_db_connection
from app.core.logging import DepLogger
from fastapi import APIRouter, HTTPException

router = APIRouter()


@router.get("/")
async def root(logger: DepLogger):
    return {"message": "Welcome to the Machine Learning API"}


@router.get("/health-check")
async def health_check(logger: DepLogger, db: DepDB):
    """
    Checks the health of the application and related connection.
    """
    logger.info("Performing health check...")

    if await check_db_connection():
        logger.info("Health check successful: Database connection verified.")
        return {"status": "ok", "database": "connected"}
    else:
        logger.error("Health check failed: Database connection error.")
        raise HTTPException(
            status_code=503,
            detail="Database connection error",
        )

File: templates/rust-ai/src/router/root.rs
//! `/` and `/health-check` routes.
//!
//! Mirrors `app/router/root.py` byte-for-byte on the response shapes:
//!
//! - `GET /` returns `200 {"message": "Welcome to the Machine Learning API"}`
//! - `GET /health-check` returns `200 {"status": "ok", "database": "connected"}`
//!   on success, or `503 {"detail": "Database connection error"}` on DB failure

use axum::Json;
use axum::extract::State;
use axum::http::StatusCode;
use axum::response::{IntoResponse, Response};
use axum::routing::get;
use serde_json::json;

use crate::AppState;

pub fn router() -> axum::Router<AppState> {
    axum::Router::new()
        .route("/", get(root))
        .route("/health-check", get(health_check))
}

async fn root() -> Json<serde_json::Value> {
    Json(json!({"message": "Welcome to the Machine Learning API"}))
}

async fn health_check(State(state): State<AppState>) -> Response {
    tracing::info!("Performing health check...");
    if state.db.check_connection().await {
        tracing::info!("Health check successful: Database connection verified.");
        (
            StatusCode::OK,
            Json(json!({"status": "ok", "database": "connected"})),
        )
            .into_response()
    } else {
        tracing::error!("Health check failed: Database connection error.");
        (
            StatusCode::SERVICE_UNAVAILABLE,
            Json(json!({"detail": "Database connection error"})),
        )
            .into_response()
    }
}
